site/src/Map.jsx

```diff
@@ -1,135 +1,238 @@
+};
+
+ThemeSource.propTypes = {
+  name: PropTypes.string.isRequired,
+  url: PropTypes.string.isRequired,
+};
+
+const ThemeTypeLayer = ({ theme, type, color, point, line, polygon }) => {
+  return (
+    <>
+      {point ? (
+        <Layer
+          beforeId="divisions_division"
+          id={`${theme}_${type}_point`}
+          type="circle"
+          source={theme}
+          source-layer={type}
+          paint={{
+            "circle-color": color,
+            "circle-radius": [
+              "interpolate",
+              ["exponential", 2],
+              ["zoom"],
+              0,
+              1,
+              17,
+              8,
+            ],
+          }}
+        />
+      ) : null}
+      {line ? (
+        <Layer
+          beforeId="divisions_division"
+          id={`${theme}_${type}_line`}
+          type="line"
+          source={theme}
+          source-layer={type}
+          paint={{ "line-color": color }}
+        />
+      ) : null}
+      {polygon ? (
+        <Layer
+          beforeId="divisions_division"
+          id={`${theme}_${type}_fill`}
+          type="fill"
+          source={theme}
+          source-layer={type}
+          paint={{ "fill-color": color, "fill-opacity": 0.25 }}
+        />
+      ) : null}
+    </>
+  );
 };
-export default function Map({mode}) {
 
-  const [pmTilesReady, setPmTilesReady] = useState(false)
-  const [cursor, setCursor] = useState('auto');
+ThemeTypeLayer.propTypes = {
+  theme: PropTypes.string.isRequired,
+  type: PropTypes.string.isRequired,
+  color: PropTypes.string.isRequired,
+  point: PropTypes.bool,
+  line: PropTypes.bool,
+  polygon: PropTypes.bool,
+};
+
+export default function Map({ mode }) {
+  const mapRef = useRef();
+  const [cursor, setCursor] = useState("auto");
   const [mapEntity, setMapEntity] = useState({});
 
+  const [visibleThemes, setVisibleThemes] = useState([]);
+  const [interactiveLayerIds, setInteractiveLayerIds] = useState([]);
+
   useEffect(() => {
-    const protocol = new pmtiles.Protocol()
-    maplibregl.addProtocol('pmtiles', protocol.tile)
-    setPmTilesReady(true)
+    const protocol = new pmtiles.Protocol();
+    maplibregl.addProtocol("pmtiles", protocol.tile);
+    return () => {
+      maplibregl.removeProtocol("pmtiles");
+    };
   }, []);
 
-  const [interactiveLayerIds, setInteractiveLayerIds] = useState(['places']);
-  // const onInteractiveLayersChange = useCallback(layerFilter => {
-  //   setInteractiveLayerIds(MAP_STYLE.layers.map(layer => layer.id).filter(layerFilter));
-  // }, []);
+  const syncInteractiveLayerIds = useCallback(() => {
+    const layers = mapRef.current.getStyle().layers;
+    const layersToShow = layers
+      .filter((layer) => visibleThemes.indexOf(layer.source) >= 0)
+      .map((layer) => layer.id);
+    setInteractiveLayerIds(layersToShow);
+  }, [visibleThemes]);
 
-  const onMouseEnter = useCallback(() => setCursor('pointer'), []);
```

Comment:
do we want prettier, biome etc for consistent style?

Reply:
One step ahead of ya :)

https://github.com/OvertureMaps/io-site/pull/55